backend/Application/Api/GraphQL/Import/DelegationImportHandler.cs

```diff
@@ -29,20 +29,20 @@ public async Task<DelegationUpdateResults> HandleDelegationUpdates(BlockDataPayl
         // meaning we can just return for blocks from a protocol version prior to that.
         if (payload.BlockInfo.ProtocolVersion < ProtocolVersion.P4) return resultBuilder.Build();
 
-        // Handle effective pending changes for delegators and update the resultBuilder.
-        if (importPaydayStatus is FirstBlockAfterPayday firstBlockAfterPayday) {
-            if (payload.BlockInfo.ProtocolVersion < ProtocolVersion.P7) {
-                // Stake changes only take effect from the first block in each payday.
-                await _writer.UpdateAccountsWithPendingDelegationChange(firstBlockAfterPayday.PaydayTimestamp,
-                    account => ApplyPendingChange(account, resultBuilder));
-            } else if (payload.BlockInfo.ProtocolVersion == ProtocolVersion.P7) {
-                // Starting from Concordium Protocol Version 7 stake changes are immediate,
-                // meaning no delegators are expected to have pending changes from this point.
-                // Only the first reward day in P7 should this do anything, afterwards it is a
-                // no-op, since no accounts with pending changes are expected.
-                await _writer.UpdateAccountsWithPendingDelegationChange(DateTimeOffset.MaxValue,
-                    account => ApplyPendingChange(account, resultBuilder));
-            }
+        if (payload.BlockInfo.ProtocolVersion < ProtocolVersion.P7 && importPaydayStatus is FirstBlockAfterPayday firstBlockAfterPayday) {
+            // Handle effective pending changes for delegators and update the resultBuilder.
+            // Stake changes only take effect from the first block in each payday.
+            await _writer.UpdateAccountsWithPendingDelegationChange(firstBlockAfterPayday.PaydayTimestamp,
+                account => ApplyPendingChange(account, resultBuilder));
+
+        } else if (payload.BlockInfo.ProtocolVersion == ProtocolVersion.P7) {
+            // Starting from Concordium Protocol Version 7 stake changes are immediate,
+            // meaning no delegators are expected to have pending changes from this point.
+            // Only the block in P7 should this do anything, afterwards it is a
+            // no-op, since no accounts with pending changes are expected.
+            await _writer.UpdateAccountsWithPendingDelegationChange(DateTimeOffset.MaxValue,
+                account => ApplyPendingChange(account, resultBuilder));
+
```

Comment:
Not a blocker, but could be:
```suggestion
        } else if (payload.BlockInfo.ProtocolVersion >= ProtocolVersion.P7) {
            // Starting from Concordium Protocol Version 7 stake changes are immediate,
            // meaning no delegators are expected to have pending changes from this point.
            // Only the block in P7 should this do anything, afterwards it is a
            // no-op, since no accounts with pending changes are expected.
            await _writer.UpdateAccountsWithPendingDelegationChange(DateTimeOffset.MaxValue,
                account => ApplyPendingChange(account, resultBuilder));

```

Also seems to reflect the comment more precisely?

Reply:
This is migrating state from P6 to P7, so when we get to P8 it does not need to be run.
Ideally we would not run it more than once, either but this was quick solution.